core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -895,6 +920,80 @@ class CivilizationInfo {
         ).toInt()
     }
 
+    fun updateProximity(otherCiv: CivilizationInfo, preCalculated: Proximity? = null): Proximity {
+        if (otherCiv == this)   return Proximity.None
+        if (preCalculated != null) {
+            // We usually want to update this for a pair of civs at the same time
+            // Since this function *should* be symmetrical for both civs, we can just do it once
+            this.proximity[otherCiv.civName] = preCalculated
+            return preCalculated
+        }
+        if (cities.isEmpty() || otherCiv.cities.isEmpty()) {
+            proximity[otherCiv.civName] = Proximity.None
+            return Proximity.None
+        }
+
+        val height = gameInfo.tileMap.mapParameters.mapSize.height
+        val width = gameInfo.tileMap.mapParameters.mapSize.width
+        var minDistance = height + width // a long distance
+        var totalDistance = 0
+        var connections = 0
+
+        var proximity = Proximity.None
+
+        for (ourCity in cities) {
+            for (theirCity in otherCiv.cities) {
+                val distance = ourCity.getCenterTile().aerialDistanceTo(theirCity.getCenterTile())
+                totalDistance += distance
+                connections++
+                if (minDistance > distance) minDistance = distance
+            }
+        }
+
+        if (minDistance <= 7) {
+            proximity = Proximity.Neighbors
+        }
+        else if (connections > 0) {
+            val averageDistance = totalDistance / connections
+            val mapFactor = (height + width) / 2 // Perhaps slightly lower for hexagonal maps??
+            val closeDistance = ((mapFactor * 25) / 100).coerceIn(10, 20)
+            val farDistance = ((mapFactor * 45) / 100).coerceIn(20, 50)
+
+            proximity = if (minDistance <= 11 && averageDistance <= closeDistance)
+                Proximity.Close
+            else if (averageDistance <= farDistance)
+                Proximity.Far
+            else
+                Proximity.Distant
+        }
+
+        // Check if different continents (unless already max distance, or water map)
+        if (connections > 0 && proximity != Proximity.Distant
+            && gameInfo.tileMap.mapParameters.type != MapType.archipelago) {
+
+            if (getCapital().getCenterTile().getContinent() != otherCiv.getCapital().getCenterTile().getContinent()) {
+                // Different continents - increase separation by one step
+                proximity = when (proximity) {
```

Comment:
.. or `operator fun inc() = values()[ordinal+1]` in the enum class and `proximity++` here .. Surprisingly, that works if you know you can't hit the 'edges' .. For short cases like this when is probably clearer.